File: frontend/src/types/procedure.ts
export interface Supply {
  supplyId: number
  quantity: number
  supplyName?: string // Thêm tên vật tư để hiển thị
}

export interface SupplyItem {
  id: number
  name: string
  unit: string
  price: number
  inStock: number
}

export interface Procedure {
  procedureId: number
  procedureName: string
  price: number
  description: string
  discount: number
  warrantyPeriod: string
  originalPrice: number
  consumableCost: number
  referralCommissionRate: number
  doctorCommissionRate: number
  assistantCommissionRate: number
  technicianCommissionRate: number
  createdAt: string
  updatedAt: string | null
  createdBy: string | null
  updatedBy: string | null
  duration?: number
  requirements?: string
  isDeleted?: boolean
  suppliesUsed?: Supply[]
}

export interface ProcedureCreateForm {
  procedureName: string
  price: number
  description: string
  discount: number
  warrantyPeriod: string
  originalPrice: number
  consumableCost: number
  referralCommissionRate: number
  doctorCommissionRate: number
  assistantCommissionRate: number
  technicianCommissionRate: number
  suppliesUsed: Supply[]
}

export interface ProcedureUpdateForm {
  procedureId: number
  procedureName: string
  price: number
  description: string
  discount: number
  warrantyPeriod: string
  originalPrice: number
  consumableCost: number
  referralCommissionRate: number
  doctorCommissionRate: number
  assistantCommissionRate: number
  technicianCommissionRate: number
  suppliesUsed?: Supply[]
}

File: backend/HolaSmileDMS/Domain/Entities/Procedure.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Procedure
{
    [Key]
    public int ProcedureId { get; set; }

    [MaxLength(200)]
    public string? ProcedureName { get; set; }
    

    [Column(TypeName = "decimal(18,2)")]
    public decimal Price { get; set; }

    [MaxLength(500)]
    public string? Description { get; set; }

    public float? Discount { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal? OriginalPrice { get; set; }
    
    [Column(TypeName = "decimal(18,2)")]
    public decimal? ConsumableCost { get; set; }

    public float? ReferralCommissionRate { get; set; }
    public float? DoctorCommissionRate { get; set; }
    public float? AssistantCommissionRate { get; set; }
    public float? TechnicianCommissionRate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public int? CreatedBy { get; set; }
    public int? UpdatedBy { get; set; }

    public bool IsDeleted { get; set; }

    public ICollection<SuppliesUsed> SuppliesUsed { get; set; }


}
